File: tennis-intelligence-platform/api/schemas/rankings.py
"""
schemas/rankings.py — Pydantic response models for the Rankings dashboard.
"""

from __future__ import annotations

from pydantic import BaseModel


class CurrentEloEntry(BaseModel):
    rank: int
    player_id: str
    player_name: str
    elo: float


class PeakEloEntry(BaseModel):
    rank: int
    player_id: str
    player_name: str
    peak_elo: float
    date_achieved: str | None


class SurfaceEloEntry(BaseModel):
    rank: int
    player_id: str
    player_name: str
    surface_elo: float


class PeakSurfaceEloEntry(BaseModel):
    rank: int
    player_id: str
    player_name: str
    peak_surface_elo: float
    date_achieved: str | None


class UpsetEntry(BaseModel):
    rank: int
    match_id: str
    date: str | None
    tournament: str
    round: str
    winner_name: str
    winner_elo: float
    loser_name: str
    loser_elo: float
    elo_gap: float

File: tennis-intelligence-platform-web/src/types/rankings.ts
// types/rankings.ts — matches api/schemas/rankings.py exactly.

export interface CurrentEloEntry {
  rank: number;
  player_id: string;
  player_name: string;
  elo: number;
}

export interface PeakEloEntry {
  rank: number;
  player_id: string;
  player_name: string;
  peak_elo: number;
  date_achieved: string | null;
}

export interface SurfaceEloEntry {
  rank: number;
  player_id: string;
  player_name: string;
  surface_elo: number;
}

export interface PeakSurfaceEloEntry {
  rank: number;
  player_id: string;
  player_name: string;
  peak_surface_elo: number;
  date_achieved: string | null;
}

export interface UpsetEntry {
  rank: number;
  match_id: string;
  date: string | null;
  tournament: string;
  round: string;
  winner_name: string;
  winner_elo: number;
  loser_name: string;
  loser_elo: number;
  elo_gap: number;
}
